backend/src/app.service.ts

```diff
@@ -67,4 +67,14 @@ export class AppService {
     const closingTime = this.lotteryContract.betsClosingTime();
     const closingTimeDate = new Date(closingTime.toNumber() * 1000);
   }
+
+  async openBets(duration: number | undefined) {
+    const privateKey = this.getPrivateKey();
+    const wallet = new ethers.Wallet(privateKey).connect(this.provider);
+    const currentBlock = this.provider.getBlock('latest');
```

Comment:
Let's check after this line if the lottery is already open, if it is - return from the function.